kiwi/boot/image/dracut.py

```diff
@@ -32,11 +32,26 @@
 from kiwi.exceptions import KiwiDiskBootImageError
 
 
+class BootNames(NamedTuple):
+    """Container for the kernel and initrd name."""
+    kernel_name: str
+    initrd_name: str
```

Comment:
I don't understand the advantage of a namedtuple return vs. a class that inherits from namedtuple and returning class variables that matches the tuple names. Is it not possible to write:

```python
def get_boot_names(self) -> namedtuple:
```

Reply:
A `namedtuple` can be absolutely anything and returning one contradicts the point of type checks. The idea is that a static type checker (like mypy or pytype) is able to tell you that this is going to fail:
``` python
boot_names = image.get_boot_names()
boot_names.this_does_not_exist = 3  # whoops AttributeError is raised here
```
and it can only know that if you specify explicitly which elements your namedtuple contains.

Furthermore, specifying this explicitly is imho a good thing, because currently we are only specifying the elements of this namedtuple in the docstring, which doesn't cause any kind of failure when someone changes the namedtuple and not the docstring.

Reply:
I agree makes sense, also you have a point here with the documentation. We would need to change this at some places in the code because I used namedtuple a lot mostly because the values are static(read-only). So many return types makes use of it